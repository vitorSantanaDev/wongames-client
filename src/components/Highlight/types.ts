export type HighlightProps = {
  title: string
  subtitle: string
  buttonLabel: string
  buttonLink: string
  backgroundImage: string
  floatImage?: string
  alignment?: 'right' | 'left'
}
