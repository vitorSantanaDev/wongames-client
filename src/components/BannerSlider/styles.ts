import styled, { css } from 'styled-components'
import * as BannerStyles from 'components/Banner/styles'
import media from 'styled-media-query'

export const Wrapper = styled.section`
  ${({ theme }) => css`
    .slick-dots {
      list-style: none;
      display: flex !important;
      align-items: center;
      justify-content: center;
      margin-top: ${theme.spacings.small};

      li {
        background-color: ${theme.colors.white};
        width: 1.2rem;
        height: 1.2rem;
        border-radius: 100%;
        display: flex;
        align-items: center;
        justify-content: center;
        margin: 0 ${theme.spacings.xxsmall};
        cursor: pointer;

        &.slick-active {
          background-color: ${theme.colors.primary};
        }
      }

      button {
        opacity: 0;
        width: 1.2rem;
        height: 1.2rem;
        cursor: pointer;
      }
    }

    ${media.greaterThan('large')`
      ${BannerStyles.Wrapper} {
        max-width: 104rem;
        margin: 0 auto;
      }

      .slick-dots {
        position: absolute;
        right: 0;
        top: 0;
        margin: 0;
        height: 100%;
        flex-direction: column;

        li {
          margin: ${theme.spacings.xxsmall} 0;
        }
      }
    `}
  `}
`
