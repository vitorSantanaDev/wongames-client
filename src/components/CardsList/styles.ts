import styled, { css } from 'styled-components'

export const Card = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;

    color: ${theme.colors.black};
    background-color: ${theme.colors.lightGray};
    padding: 1.3rem ${theme.spacings.xsmall};

    &:not(:last-child) {
      margin-bottom: ${theme.spacings.xsmall};
    }

    > span {
      margin-left: ${theme.spacings.xxsmall};
    }
  `}
`
