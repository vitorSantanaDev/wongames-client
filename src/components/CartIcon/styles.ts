import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    width: 2.4rem;
    height: 2.4rem;
    color: ${theme.colors.white};
    position: relative;
  `}
`

export const Badge = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    color: ${theme.colors.white};
    background-color: ${theme.colors.secondary};

    font-size: 1rem;
    border-radius: 50%;
    width: 1.6rem;
    height: 1.6rem;
    position: absolute;
    top: -0.4rem;
    right: -0.4rem;
  `}
`
