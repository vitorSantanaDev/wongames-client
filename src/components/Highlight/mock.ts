export default {
  buttonLink: '/rdl2',
  buttonLabel: 'Buy now',
  title: "Read Dead it's back",
  backgroundImage: '/img/red-dead-img.png',
  subtitle: "Come see John's new adventures"
}
