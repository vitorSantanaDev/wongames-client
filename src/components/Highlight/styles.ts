import styled, { css } from 'styled-components'
import media from 'styled-media-query'
import { HighlightProps } from './types'

type WrapperProps = Pick<HighlightProps, 'backgroundImage' | 'alignment'>

const wrapperModifiers = {
  right: () => css`
    grid-template-columns: 1.3fr 2fr;
    grid-template-areas: 'floatImage content';
    ${Content} {
      text-align: right;
    }
  `,
  left: () => css`
    grid-template-columns: 2fr 1.3fr;
    grid-template-areas: 'content floatImage';
    ${Content} {
      text-align: left;
    }

    ${FloatImage} {
      justify-self: end;
    }
  `
}

export const Wrapper = styled.section<WrapperProps>`
  ${({ backgroundImage, alignment }) => css`
    position: relative;
    height: 23rem;
    background-size: cover;
    background-position: center center;
    background-image: url(${backgroundImage});

    display: grid;
    ${wrapperModifiers[alignment!]()}

    &::after {
      content: '';
      width: 100%;
      height: 100%;
      position: absolute;
      background-color: rgba(0, 0, 0, 0.6);
    }

    ${media.greaterThan('medium')`
      height: 32rem;
    `}
  `}
`

export const FloatImage = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    max-height: 23rem;
    align-self: end;
    z-index: ${theme.layers.base};

    grid-area: floatImage;

    ${media.greaterThan('medium')`
      max-height: 32rem;
    `}
  `}
`

export const Content = styled.div`
  ${({ theme }) => css`
    padding: ${theme.spacings.xsmall};
    text-align: right;
    z-index: ${theme.layers.base};

    grid-area: content;

    ${media.greaterThan('medium')`
      align-self: end;
      padding: ${theme.spacings.large};
    `}
  `}
`

export const Title = styled.h2`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    font-weight: ${theme.font.bold};
    font-size: ${theme.font.sizes.large};

    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.xxlarge};
    `}
  `}
`

export const Subtitle = styled.h3`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    font-weight: ${theme.font.light};
    font-size: ${theme.font.sizes.small};
    margin-bottom: ${theme.spacings.medium};

    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.large};
    `}
  `}
`
