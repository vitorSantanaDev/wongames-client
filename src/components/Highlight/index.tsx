import Button from 'components/Button'
import * as S from './styles'

import { HighlightProps } from './types'
import Image from 'next/image'

const Highlight = ({
  subtitle,
  title,
  buttonLink,
  buttonLabel,
  backgroundImage,
  floatImage,
  alignment = 'right'
}: HighlightProps) => (
  <S.Wrapper alignment={alignment}>
    <Image src={backgroundImage} alt={title} layout="fill" />
    {!!floatImage && (
      <S.FloatImageWrapper>
        <Image
          width={400}
          height={300}
          src={floatImage}
          alt={`Float image ${title}`}
        />
      </S.FloatImageWrapper>
    )}
    <S.Content>
      <S.Title>{title}</S.Title>
      <S.Subtitle>{subtitle}</S.Subtitle>
      <Button as="a" href={buttonLink}>
        {buttonLabel}
      </Button>
    </S.Content>
  </S.Wrapper>
)

export default Highlight
