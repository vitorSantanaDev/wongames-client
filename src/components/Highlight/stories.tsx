import { Story, Meta } from '@storybook/react'
import Highlight from '.'
import { HighlightProps } from './types'

export default {
  title: 'Highlight',
  component: Highlight,
  args: {
    buttonLink: '/rdl2',
    buttonLabel: 'Buy now',
    title: "Read Dead it's back",
    backgroundImage: '/img/red-dead-img.png',
    subtitle: "Come see John's new adventures"
  }
} as Meta

export const Default: Story<HighlightProps> = (args) => (
  <div style={{ maxWidth: '104rem' }}>
    <Highlight {...args} />
  </div>
)

export const WithFloatImage: Story<HighlightProps> = (args) => (
  <div style={{ maxWidth: '104rem' }}>
    <Highlight {...args} />
  </div>
)

WithFloatImage.args = {
  floatImage: '/img/red-dead-float.png'
}
