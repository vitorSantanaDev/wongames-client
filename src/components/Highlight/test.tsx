import { render, screen } from 'utils/test-utils'
import Highlight from '.'
import * as S from './styles'

const props = {
  title: 'Heading 1',
  subtitle: 'Heading 2',
  buttonLabel: 'Buy now',
  buttonLink: '/rdl2',
  backgroundImage: '/img/read-dead-img.png'
}

describe('<Highlight />', () => {
  it('should render headings and button', () => {
    const { container } = render(<Highlight {...props} />)

    expect(
      screen.getByRole('heading', { name: /heading 1/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /heading 2/i })
    ).toBeInTheDocument()

    expect(screen.getByRole('link', { name: /buy now/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render background image', () => {
    const { container } = render(<Highlight {...props} />)

    expect(container.firstChild).toHaveStyle({
      'background-image': `url(${props.backgroundImage})`
    })
  })

  it('should render background image', () => {
    render(<Highlight floatImage="/img/float-image.png" {...props} />)

    expect(screen.getByRole('img', { name: props.title })).toHaveAttribute(
      'src',
      '/img/float-image.png'
    )
  })

  it('should render float image align right by default', () => {
    const { container } = render(<Highlight {...props} />)

    expect(container.firstChild).toHaveStyleRule(
      'grid-template-areas',
      "'floatImage content'"
    )
    expect(container.firstChild).toHaveStyleRule('text-align', 'right', {
      modifier: `${S.Content}`
    })
  })

  it('should render float image align left when prop is passed', () => {
    const { container } = render(<Highlight alignment="left" {...props} />)

    expect(container.firstChild).toHaveStyleRule(
      'grid-template-areas',
      "'content floatImage'"
    )
    expect(container.firstChild).toHaveStyleRule('text-align', 'left', {
      modifier: `${S.Content}`
    })
  })
})
