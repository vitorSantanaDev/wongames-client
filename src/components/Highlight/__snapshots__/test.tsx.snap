// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Highlight /> should render headings and button 1`] = `
.c5 {
  border: none;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #FAFAFA;
  padding: 0.8rem;
  border-radius: 0.4rem;
  background: linear-gradient(180deg,#ff605f 0%,#f062c0 50%);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 4rem;
  font-size: 1.4rem;
  padding: 0.8rem 3.2rem;
}

.c5:focus {
  outline: 1px dashed currentColor;
}

.c5:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c0 {
  position: relative;
  height: 23rem;
  display: grid;
  grid-template-columns: 1.3fr 2fr;
  grid-template-areas: 'floatImage content';
}

.c0 .c1 {
  text-align: right;
}

.c0 img {
  position: absolute;
  object-fit: cover;
}

.c0::after {
  content: '';
  width: 100%;
  height: 100%;
  position: absolute;
  background-color: rgba(0,0,0,0.6);
}

.c2 {
  padding: 1.6rem;
  text-align: right;
  z-index: 10;
  grid-area: content;
}

.c3 {
  color: #FAFAFA;
  font-weight: 600;
  font-size: 1.8rem;
}

.c4 {
  color: #FAFAFA;
  font-weight: 300;
  font-size: 1.4rem;
  margin-bottom: 3.2rem;
}

@media (min-width:768px) {
  .c0 {
    height: 32rem;
  }
}

@media (min-width:768px) {
  .c2 {
    -webkit-align-self: end;
    -ms-flex-item-align: end;
    align-self: end;
    padding: 4.0rem;
  }
}

@media (min-width:768px) {
  .c3 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c4 {
    font-size: 1.8rem;
  }
}

<section
  class="c0"
>
  <img
    alt="Heading 1"
    layout="fill"
    src="/img/read-dead-img.png"
  />
  <div
    class="c1 c2"
  >
    <h2
      class="c3"
    >
      Heading 1
    </h2>
    <h3
      class="c4"
    >
      Heading 2
    </h3>
    <a
      class="c5"
      href="/rdl2"
    >
      <span>
        Buy now
      </span>
    </a>
  </div>
</section>
`
